apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java

```diff
@@ -190,8 +202,20 @@ private void addRandomFieldValues(FissClaim.Builder claim) {
           claim.setMedaProv6(medaProvId.substring(0, 6));
         });
     optional("totalChargeAmount", () -> claim.setTotalChargeAmount(randomAmount()));
-    optional("recdDtCymd", () -> claim.setRecdDtCymd(randomDate()));
-    optional("currTranDtCymd", () -> claim.setCurrTranDtCymd(randomDate()));
+    optional(
+        "recdDtCymd",
+        () -> {
+          String date = randomDate();
+          claim.setRecdDtCymd(date);
+          claim.setRecdDtCymdText(date);
```

Comment:
In the real data stream is `recvd_dt_cymd` null and `recd_dt_cymd_text` populated when the text is an invalid date?  To keep the behavior that the two match on a valid date maybe use a `oneOf` for this with lambdas that set either just the text (invalid date case) or both (valid date case)?

Reply:
Looked into this, and it seems on previous fields where a `_text` value was added, both values are sent.